pfcpiface/p4rt_translator_test.go

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Open Networking Foundation
+
+package pfcpiface
+
+import (
+	"io/ioutil"
+	"testing"
+
+	//nolint:staticcheck // Ignore SA1019.
+	// Upgrading to google.golang.org/protobuf/proto is not a drop-in replacement,
+	// as also P4Runtime stubs are based on the deprecated proto.
+	"github.com/golang/protobuf/proto"
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	p4InfoTestPath = "../conf/p4/bin/p4info.txt"
+
+	mockImmutableP4Info = "tables {\n  preamble {\n    id: 33586128\n    name: \"decap_cpu_header\"\n    alias: \"decap_cpu_header\"\n  }\n  action_refs {\n    id: 16788917\n  }\n  size: 1024\n}\ntables {\n  preamble {\n    id: 33589124\n    name: \"forward\"\n    alias: \"forward\"\n  }\n  match_fields {\n    id: 1\n    name: \"routing_metadata.nhop_ipv4\"\n    bitwidth: 32\n    match_type: EXACT\n  }\n  action_refs {\n    id: 16780303\n  }\n  action_refs {\n    id: 16840314\n  }\n  action_refs {\n    id: 16784184\n  }\n  size: 512\n}\ntables {\n  preamble {\n    id: 33581985\n    name: \"ipv4_lpm\"\n    alias: \"ipv4_lpm\"\n  }\n  match_fields {\n    id: 1\n    name: \"ipv4.dstAddr\"\n    bitwidth: 32\n    match_type: LPM\n  }\n  action_refs {\n    id: 16812204\n  }\n  action_refs {\n    id: 16784184\n  }\n  size: 1024\n}\ntables {\n  preamble {\n    id: 33555613\n    name: \"send_arp_to_cpu\"\n    alias: \"send_arp_to_cpu\"\n  }\n  action_refs {\n    id: 16840314\n  }\n  size: 1024\n}\ntables {\n  preamble {\n    id: 33562826\n    name: \"send_frame\"\n    alias: \"send_frame\"\n  }\n  match_fields {\n    id: 1\n    name: \"standard_metadata.egress_port\"\n    bitwidth: 9\n    match_type: EXACT\n  }\n  action_refs {\n    id: 16813016\n  }\n  action_refs {\n    id: 16784184\n  }\n  size: 256\n}\nactions {\n  preamble {\n    id: 16788917\n    name: \"do_decap_cpu_header\"\n    alias: \"do_decap_cpu_header\"\n  }\n}\nactions {\n  preamble {\n    id: 16780303\n    name: \"set_dmac\"\n    alias: \"set_dmac\"\n  }\n  params {\n    id: 1\n    name: \"dmac\"\n    bitwidth: 48\n  }\n}\nactions {\n  preamble {\n    id: 16840314\n    name: \"do_send_to_cpu\"\n    alias: \"do_send_to_cpu\"\n  }\n  params {\n    id: 1\n    name: \"reason\"\n    bitwidth: 16\n  }\n  params {\n    id: 2\n    name: \"cpu_port\"\n    bitwidth: 9\n  }\n}\nactions {\n  preamble {\n    id: 16784184\n    name: \"_drop\"\n    alias: \"_drop\"\n  }\n}\nactions {\n  preamble {\n    id: 16812204\n    name: \"set_nhop\"\n    alias: \"set_nhop\"\n  }\n  params {\n    id: 1\n    name: \"nhop_ipv4\"\n    bitwidth: 32\n  }\n  params {\n    id: 2\n    name: \"port\"\n    bitwidth: 9\n  }\n}\nactions {\n  preamble {\n    id: 16813016\n    name: \"rewrite_mac\"\n    alias: \"rewrite_mac\"\n  }\n  params {\n    id: 1\n    name: \"smac\"\n    bitwidth: 48\n  }\n}"
```

Comment:
What's the need for two p4info files?

Reply:
the purpose was to test basic methods like `actionID()` on a different p4info file.

After a brief discussion we agreed on using auto generated p4 constants (e.g. [Constants.java](https://github.com/stratum/fabric-tna/blob/main/src/main/java/org/stratumproject/fabric/tna/Constants.java), in fabric-tna).